### PR TITLE
5 create base models

### PR DESCRIPTION
**Pull request by radthenone**
comment to review

### CODE REVIEW

File: backend/apps/forum/tests/test_models.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    @pytest.fixture()
+    def reply_fifty_model_id(self) -> List[None | Reply]:
+        reply_parent = ReplyFactory.create(parent=None)
+        for _ in range(50):
+            ReplyFactory.create(parent=reply_parent)
+        return reply_parent.id
+
+    def test_reply_parent_sum(self, reply_fifty_model_id):
+        assert Reply.objects.sum_replies(id=reply_fifty_model_id) == 50
```
Comment (Indoxer) [inline backend/apps/forum/tests/test_models.py:99]: You can also test \_\_str\_\_ methods, regex validators (It is django functionality but your regex not)

**Review by Indoxer — CHANGES_REQUESTED** [review]

File: backend/apps/forum/models.py [hunk 2]
```diff
@@ -1,4 +1,163 @@
...
+        post (ForeignKey): A foreign key to the Post.
+    """
+
+    content = MartorField()
+    created_at = models.DateTimeField(
+        verbose_name="Reply created",
+        auto_now_add=True,
+    )
+    updated_at = models.DateTimeField(
+        verbose_name="Reply updated",
```
Comment (Indoxer) [inline backend/apps/forum/models.py:119]: -||-

File: backend/apps/forum/models.py [hunk 3]
```diff
@@ -1,4 +1,163 @@
...
+        validators=[
+            RegexValidator(
+                regex=r"^[a-zA-Z ]+$",
+                message="Input must contain only alphabetic characters",
+            ),
+        ],
+    )
+    body = MartorField()
+    created_at = models.DateTimeField(
+        auto_now_add=True,
```
Comment (Indoxer) [inline backend/apps/forum/models.py:66]: You can add verbose_name

File: backend/apps/forum/models.py [hunk 4]
```diff
@@ -1,4 +1,163 @@
...
+    )
+
+    objects = PostManager()
+
+    def __str__(self):
+        """Return the string representation of the model."""
+        return self.title
+
+    def save(self, *args, **kwargs):
+        self.body = markdown_text(self.body)
```
[the remark this answers is not in the document]
Comment (Indoxer) [reply backend/apps/forum/models.py:88]: not html

File: backend/apps/forum/tests/test_models.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+            category=CategoryFactory.create(name="category"),
+            author=UserFactory.create(
+                username="emil123",
+                email="emil123@example.com",
+                first_name="Emil",
+                last_name="Dostowski",
+            ),
+        )
+
+    def test_post_model(self, post_model) -> None:
```
[the remark this answers is not in the document]
Comment (Indoxer) [reply backend/apps/forum/tests/test_models.py:66]: Testing everything may be an option (in pure TDD it is required), but I think we can make our lives easier and test things that can break easily

File: backend/apps/forum/apps.py [hunk 6]
```diff
@@ -1,6 +1,14 @@
...
 
 
 class ForumConfig(AppConfig):
     default_auto_field = "django.db.models.BigAutoField"
     name = "backend.apps.forum"
+
+    def ready(self):
+        from backend.apps.forum import signals
+
+        request_finished.connect(signals.category_name)
```
[the remark this answers is not in the document]
Comment (radthenone) [reply backend/apps/forum/apps.py:12]: ok i removed request_finished lines.

[the remark this answers is not in the document]
Comment (radthenone) [reply backend/apps/forum/apps.py:12]: ok only    `    def ready(self):
        from backend.apps.forum import signals` stay here

File: backend/apps/forum/managers.py [hunk 7]
```diff
@@ -0,0 +1,11 @@
+from django.db import models
+
+
+class PostManager(models.Manager):
+    def sum_posts(self):
+        return len(super().all())
+
+
+class ReplyManager(models.Manager):
```
[the remark this answers is not in the document]
Comment (radthenone) [reply backend/apps/forum/managers.py:9]: Post now use count but work with manager, Reply now used @property method

File: backend/apps/forum/models.py [hunk 8]
```diff
@@ -1,4 +1,163 @@
...
+
+    name = models.CharField(
+        max_length=100,
+        blank=False,
+        unique=True,
+        verbose_name="Category name",
+        validators=[
+            RegexValidator(
+                regex=r"^[a-zA-Z]*$",
+                message="Input must contain only alphabetic characters",
```
[the remark this answers is not in the document]
Comment (radthenone) [reply backend/apps/forum/models.py:29]: done, ^[a-zA-Z]+$

File: backend/apps/forum/models.py [hunk 9]
```diff
@@ -1,4 +1,163 @@
...
+            ),
+        ],
+    )
+
+    def __str__(self):
+        """Return the string representation of the model."""
+        return self.name
```
[the remark this answers is not in the document]
Comment (radthenone) [reply backend/apps/forum/models.py:36]: ok i added this with Meta class in all models

File: backend/apps/forum/managers.py [hunk 10]
```diff
@@ -0,0 +1,10 @@
+from django.db import models
+
+
+class PostManager(models.Manager):
+    def sum_posts(self):
```
Comment (Indoxer) [inline backend/apps/forum/managers.py:5]: This is unnecessary, to much abstraction

File: backend/apps/forum/tests/factories.py [hunk 11]
```diff
@@ -0,0 +1,48 @@
+from factory import Faker, LazyFunction, Sequence, SubFactory
+from factory.django import DjangoModelFactory
+from faker import Faker as Fake
+from mdgen import MarkdownPostProvider
+
+from backend.apps.forum.models import Category, Post, Reply
+from backend.apps.forum.utils import get_content_html
+from backend.apps.users.tests.factories import UserFactory
+
+fake = Fake()
```
Comment (Indoxer) [inline backend/apps/forum/tests/factories.py:10]: In my opinion currently is little mishmash here, you use sometimes raw Faker and sometimes Faker wrapper from factory boy

File: backend/apps/forum/models.py [hunk 12]
```diff
@@ -1,4 +1,202 @@
...
+
+    def save(self, *args, **kwargs):
+        self.content_html = get_content_html(self.content_html, self.content_markdown)
+        super(Post, self).save(*args, **kwargs)
+
+    @property
+    def get_slug_datetime(self):
+        return str(self.created_at.strftime("%H:%M:%S:%d:%m:%Y")).replace(":", "")
+
+    def slugify_function(self, content):
```
Comment (Indoxer) [inline backend/apps/forum/models.py:120]: Death code, you don't use this, AutoSlugField has built-in slugify function

File: backend/apps/forum/tests/factories.py [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+
+
+class PostFactory(DjangoModelFactory):
+    class Meta:
+        model = Post
+
+    title = fake_unique_two_words
+    content_markdown = fake_markdown
+    content_html = get_content_html(None, fake_markdown)
+    slug = title
```
Comment (Indoxer) [inline backend/apps/forum/tests/factories.py:31]: It is not valid slug, you can use LazyAttribute or PostGenerationMethodCall and built-in function to slugify from django extension

File: backend/apps/forum/utils.py [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+    return instance
+
+
+def markdown_to_html(instance: MartorField) -> str:
+    """Create html from markdown field"""
+    new_list = []
+    if instance:
+        text_list = instance.splitlines()
+        for text in text_list:
+            mark_text = str(markdowner.convert(text))
```
Comment (Indoxer) [inline backend/apps/forum/utils.py:27]: This library cannot convert many lines?

**Review by Indoxer — CHANGES_REQUESTED** [review]

**Review by Indoxer — APPROVED** [review]

**Pull request merged**